packages/jest-worker/src/index.js

```diff
@@ -68,6 +68,7 @@ export default class {
     // Build the options once for all workers to avoid allocating extra objects.
     const workerOptions = {
       forkOptions: options.forkOptions || {},
+      maxRetries: options.maxRetries || Infinity,
```

Comment:
I'm not sure Infinity is a sensible default. Why would you ever want to retry more than say 3 or 5 times? Can we set it to 3 or 5? People can still pass "Infinity" if they want, but the default should be to crash after a few retries.